### ISSUE TITLE
Error: Singular matrix in MatrixInvert

### ISSUE THREAD

**Issue by wiesehahn** [tylermorganwall/rayrender#53]
I have the issue that a scene is not rendered via `render_highquality()` when certain points are added. The error was posted at https://github.com/tylermorganwall/rayshader/issues/315 already but as it happens with highquality and not with snapshot I guess it is related to rayrender.

The Error I receive is `Error: Singular matrix in MatrixInvert`.  I am not good at math anymore but it seems there are problems to invert some matrices of the data, however I cannot see why.

Here is the same example as in the other issue but a little simplified and reduced:

(I am using latest versions of rayshader and rayrender from Github on a windows maschine)

```r
library(lidR)
library(rayshader)

f <- system.file("extdata", "Megaplot.laz", package="lidR")
las <- readLAS(f, select="xyzc")

# create DTM
dtm <- rasterize_terrain(las, 0.25, tin(), pkg = "raster")

# init 3D plot with DTM
elmat <- raster_to_matrix(dtm)
elmat %>%
  sphere_shade() %>%
  plot_3d(elmat) 

# with this data it does not work
las@data <- las@data[276:300]

# with this data it does work
# las@data <- las@data[250:275]

# add points
render_points(extent = attr(dtm,"extent"),
              lat = las@data$Y ,
              long = las@data$X ,
              altitude = las@data$Z)

# render
render_highquality(samples = 5,
                   width = 500,
                   height = 500)
```

**Comment by wiesehahn** [comment]
Probably this has something to do with data precision and correlation, I dont know. 
In the case above we can overcome the problem when we replace `long = las@data$X` with `long = as.integer(las@data$X * 10) / 10`, however `long = round(las@data$X, 1)` does not work and also `long = as.integer(las@data$X * 10) / 10` only works for the data subset (points 276-300) and not if we want to render the entire dataset.

**Comment by tylermorganwall** [comment]
This was a rayshader issue that was fixed in 368457a36bcb4955a2b6c0f6e580cc1507d65642.
